src/arch/x86/tlb.cc

```diff
@@ -371,7 +377,15 @@ TLB::translate(const RequestPtr &req,
         if (m5Reg.paging) {
             DPRINTF(TLB, "Paging enabled.\n");
             // The vaddr already has the segment base applied.
-            TlbEntry *entry = lookup(vaddr);
+
+            //Appending the pcid (last 12 bits of CR3) to the
+            //page aligned vaddr
+            Process *p_temp = tc->getProcessPtr();
+            Addr alignedVaddr_temp = p_temp->pTable->pageAlign(vaddr);
+            alignedVaddr_temp = concAddrPid(alignedVaddr_temp,
+                                tc->readMiscRegNoEffect(MISCREG_CR3) & 0xfff);
```

Comment:
I have added some updates to use CR4's pcide bit (not sure if what I am doing is the best way to do it). Sorry, I am not exactly sure about the change needed on src/arch/x86/pagetable_walker.cc:587

Reply:
Since I am only tweaking virtual address when an entry is added to the tlb or looked in the tlb, I am not sure if any change is needed on src/arch/x86/pagetable_walker.cc:587